### PR TITLE
Feature-flag query builder, off by default

### PR DESCRIPTION
**Pull request by lguychard**
Context: https://sourcegraph.slack.com/archives/CHEKCRWKV/p1602776317260800

### CODE REVIEW

File: client/web/src/search/input/SearchPage.tsx [hunk 1]
```diff
@@ -8,6 +8,7 @@ import {
     RepogroupHomepageProps,
     OnboardingTourProps,
     HomePanelsProps,
+    ShowQueryBuilderProps,
```
Comment (limitedmage) [inline client/web/src/search/input/SearchPage.tsx:11]: Do we want to add some storybook tests for when this feature is on?

**Comment by limitedmage** [comment]
Looks like removing the query builder link messes a little bit the the spacing on the homepage and repogroup pages and moves the content too close to the search bar. We may need to add some extra spacing here. FYI @rrhyne 

image

image

image

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/input/SearchPage.tsx (hunk 1 above)
Comment (lguychard) [reply client/web/src/search/input/SearchPage.tsx:11]: Done in https://github.com/sourcegraph/sourcegraph/pull/14779/commits/c6844630b3721055f1eb771b5e6affc684557d74

File: client/web/src/search/input/SearchPage.test.tsx [hunk 2]
```diff
@@ -54,6 +54,8 @@ describe('SearchPage', () => {
         showRepogroupHomepage: false,
         showEnterpriseHomePanels: false,
         showOnboardingTour: false,
+
```
Comment (limitedmage) [inline client/web/src/search/input/SearchPage.test.tsx:57]: **nit:** remove blank line

**Review by limitedmage — APPROVED** [review]

File: client/web/src/search/input/SearchPage.test.tsx (hunk 2 above)
Comment (lguychard) [reply client/web/src/search/input/SearchPage.test.tsx:57]:
```suggestion
```

**Comment by rrhyne** [comment]
The query bar is 5 rems above the content with the 'Query builder' link. The builder line occupies the space, more than it creates a new line, so I would like to maintain the 5rems between content and input when the query builder link is not present.

In what conditions is the query builder link not present? Can you still create have the search scopes there with the proper feature flag combination?

**Comment by limitedmage** [comment]
> The query bar is 5 rems above the content with the 'Query builder' link. The builder line occupies the space, more than it creates a new line, so I would like to maintain the 5rems between content and input when the query builder link is not present.

I'll add a 5rem spacing regardless of what features are on. I think changing the spacing depending on whether query builder/scopes/quicklinks/interactive search are enabled would complicate things too much.

> In what conditions is the query builder link not present? Can you still create have the search scopes there with the proper feature flag combination?

Search scopes are gone from the homepage entirely (not feature flagged), but we still have Quick Links.

Here's how it looks like with the 5rem spacing @rrhyne:

Both query builder and quick links on:
image

Only quick links:
image

Neither:
image

**Pull request merged**